.github/workflows/packages-build-macos-agent.yml

```
@@ -0,0 +1,108 @@
+name: Build Macos Wazuh Agent Packages - intel64|arm64
+on:
+  workflow_dispatch:
+    inputs:
+      architecture:
+        type: choice
+        description: Package architecture [intel64, arm64].
+        options:
+          - intel64
+          - arm64
+        required: true
+        default: 'intel64'
+      revision:
+        description: |
+          Package revision (name and metadata).
+          Default is '0'.
+        required: false
+        default: "0"
+      is_stage:
+        type: boolean
+        description: |
+          Should set development/production nomenclature
+          True if package name should have production format.
+          False if package name should have developer format.
+          Default is 'false'.
+        required: false
+        default: false
+      debug:
+        type: boolean
+        description: |
+          Activate debug mode in the compilation.
+          Default is 'false'.
+        required: false
+        default: false
+      checksum:
+        type: boolean
+        description: Generate checksum on the same directory than the package.
+        required: false
+        default: false
+
+  workflow_call:
+    inputs:
+      architecture:
+        type: string
+        required: true
+      revision:
+        type: string
+        required: false
+      is_stage:
+        type: boolean
+        required: false
+      debug:
+        type: boolean
+        required: false
+      checksum:
+        type: boolean
+        required: false
+
+jobs:
+  Build-agent-linux-packages:
+    runs-on: macos-latest
+    name: Build MacOS wazuh-agent on ${{ inputs.architecture }}
+
+    steps:
+      - name: Cancel previous runs
+        uses: fkirc/skip-duplicate-actions@master
+        with:
+          cancel_others: 'true'
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          skip_after_successful_duplicate: 'false'
+
+      - uses: actions/checkout@v4
+
+      - name: Compile Wazuh agent and generate package
+        working-directory: packages/macos/
+        run: |
+          nproc=$(sysctl -n hw.ncpu)
+          FLAGS="-j $nproc -r $REVISION "
+          if [ "${{ inputs.is_stage }}" == "true" ]; then FLAGS+="--is_stage "; fi
+          if [ "${{ inputs.checksum }}" == "true" ]; then FLAGS+="--checksum "; fi
+          if [ "${{ inputs.debug}}" == "true" ]; then FLAGS+="--debug "; fi
+          sudo bash generate_wazuh_packages.sh -a ${{ inputs.architecture }} $FLAGS
+
+      - name: Test install Wazuh agent
+        working-directory: packages/macos/output
+        run: |
+          sudo echo "WAZUH_MANAGER='1.1.1.1'" > /tmp/wazuh_envs && sudo installer -pkg *agent*pkg -target /
+          sudo /Library/Ossec/bin/wazuh-control start
+          if [ $(sudo /Library/Ossec/bin/wazuh-control status | grep 'is running...' | wc -l) -gt "0" ];
```

```suggestion
          if [ $(sudo /Library/Ossec/bin/wazuh-control status | grep 'is running...' | wc -l) -gt 5 ];
```
So, we validate that all services are running:

```
wazuh-modulesd is running...
wazuh-logcollector is running...
wazuh-syscheckd is running...
wazuh-agentd is running...
wazuh-execd is running...
```